### PR TITLE
Provide custom format for dailynotes file name

### PR DESCRIPTION
**Pull request by AntoineBalaine**
Hello, 

I created an option to provide a custom date format to daily notes, in the config. 

I love every bit of obsidian.nvim but found that I had been using a different file format, so here it is. 
 
If you like this option, let me know and I can add a small mention in the readme. 

Cheers!

### CODE REVIEW

**Comment by epwalsh** [comment]
Hey @AntoineBalaine, this is great, thank you! Please add to the README

**Comment by AntoineBalaine** [comment]
@epwalsh Et voilà!

File: lua/obsidian/config.lua [hunk 1]
```diff
@@ -52,13 +52,15 @@ end
 
 ---@class obsidian.config.DailyNotesOpts
 ---@field folder string|?
+---@field date_format string|osdate
```
Comment (epwalsh) [inline lua/obsidian/config.lua:55]:
```suggestion
---@field format_date function|?
```

File: README.md [hunk 2]
```diff
@@ -161,6 +161,22 @@ require("obsidian").setup({
 })
 ```
 
+#### Customizing daily notes' date format
+
+You can customize the file name of your daily notes by providing a date format:
```
Comment (epwalsh) [inline README.md:166]:
```suggestion
You can customize the file name of your daily notes by providing a date formatting function:
```

File: README.md [hunk 3]
```diff
@@ -161,6 +161,22 @@ require("obsidian").setup({
...
+
+```lua
+require("obsidian").setup({
+  dir = "~/my-vault",
+  daily_notes = {
+    date_format = function()
```
Comment (epwalsh) [inline README.md:172]:
```suggestion
    format_date = function()
```

File: lua/obsidian/config.lua [hunk 4]
```diff
@@ -52,13 +52,15 @@ end
...
 config.DailyNotesOpts = {}
 
 ---Get defaults.
 ---@return obsidian.config.DailyNotesOpts
 config.DailyNotesOpts.default = function()
   return {
     folder = nil,
+    date_format = nil,
```
Comment (epwalsh) [inline lua/obsidian/config.lua:63]:
```suggestion
    format_date = nil,
```

File: lua/obsidian/init.lua [hunk 5]
```diff
@@ -257,7 +257,13 @@ end
...
   ---@type string
   ---@diagnostic disable-next-line: assign-type-mismatch
-  local id = os.date "%Y-%m-%d"
+  local date_format
+  if self.opts.daily_notes.date_format ~= nil then
+    date_format = self.opts.daily_notes.date_format()
+  else
+    date_format = os.date("%Y-%m-%d")
+  end
+  local id = date_format
```
Comment (epwalsh) [inline lua/obsidian/init.lua:266]:
```suggestion
  local formatted_date
  if self.opts.daily_notes.format_date ~= nil then
    formatted_date = self.opts.daily_notes.format_date()
  else
    formatted_date = os.date("%Y-%m-%d")
  end
  local id = format_date
```

**Review by epwalsh — COMMENTED** [review]
Sorry for the slow responses..

I think we should call this field `format_date` instead of `date_format` since it's a function. What do you think?

**Comment by AntoineBalaine** [comment]
> Hi, can I continue this PR?

Of course!

**Comment by evtuhovich** [comment]
:+1:

I've monkey patched this for myself, so waiting for this PR

**Comment by AntoineBalaine** [comment]
> +1
> 
> I've monkey patched this for myself, so waiting for this PR

Here's the update.

File: lua/obsidian/init.lua [hunk 6]
```diff
@@ -266,7 +266,13 @@ end
 client.today = function(self)
   ---@type string
   ---@diagnostic disable-next-line: assign-type-mismatch
-  local id = os.date "%Y-%m-%d"
+  local formatted_date
+  if self.opts.daily_notes.format_date ~= nil then
+    formatted_date = self.opts.daily_notes.format_date()
```
Comment (epwalsh) [inline lua/obsidian/init.lua:271]: We'll need to use this below in `client.yesterday` as well. So the `format_date()` function should probably take an optional `time` argument just like `os.date()`.

**Review by epwalsh — CHANGES_REQUESTED** [review]
Thanks @AntoineBalaine, just one comment

**Comment by epwalsh** [comment]
Let's go with #138 instead. It keeps things simpler and handles the "yesterday" case as well.

**Pull request closed without merging**